core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -72,16 +77,19 @@ private void analyze(ActivityPayload payload, UserAnonymizedDTO userAnonymized,
 	private void addOrUpdateActivity(ActivityPayload payload, UserAnonymizedDTO userAnonymized, Goal matchingGoal)
```

Comment:
This method is growing too long. Please move the branches if the if/elsif into separate methods.